magma/lib/magma/gnomon/identifier.rb

```diff
@@ -13,6 +13,12 @@ def to_hash
         }
       end
 
+      def is_up_to_date?
+        most_recent_grammar = Magma::Gnomon::Grammar.for_project(project_name).to_hash
+        identifier_grammar = Magma.instance.db[:grammars].where(project_name: project_name, id: grammar_id).first
+        most_recent_grammar[:version_number] == identifier_grammar[:version_number]
```

Comment:
could avoid another db call here by doing most_recent_grammar.id == grammar_id - wish the grammar could also be memoized somehow